virtwho/register.py

```diff
@@ -42,25 +42,30 @@ def register(self):
         """
         Register host by subscription-manager command
         """
-        self.unregister()
-        cmd = f'subscription-manager register ' \
-              f'--serverurl={self.server}:{self.port}{self.prefix} ' \
-              f'--org={self.org} '
-        if self.activation_key:
-            cmd += f'--activationkey={self.activation_key} '
-        else:
-            cmd += f'--username={self.username} ' \
-                   f'--password={self.password} '
-        if self.register_type == 'satellite':
-            self.satellite_cert_install()
-        else:
-            cmd += f'--baseurl={self.baseurl}'
-        ret, output = self.ssh.runcmd(cmd)
-        if ret == 0 and 'The system has been registered' in output:
-            logger.info(f'Succeeded to register host')
-            return output
+        if not self.is_register():
```

Comment:
will try to check the register status before register a host or guest, if it already been registered in the target account, will not register again to save time.